### PR TITLE
Update process function for RouteResponse.

### PR DESCRIPTION
**Pull request by siarheishut**
Fix RenderRoute test.
Add key `map' in output map of RouteResponse if route is found. Update tests accordingly.

### CODE REVIEW

File: src/request_processor.cpp [hunk 1]
```diff
@@ -140,7 +142,12 @@ json::Dict Processor::Process(const GetStopRequest &request) const {
 }
 
 json::Dict Processor::Process(const GetRouteRequest &request) const {
-  return ToJson(bus_manager_->GetRoute(request.from, request.to), request.id);
+  auto route_info = bus_manager_->GetRoute(request.from, request.to);
+  if (!route_info.has_value())
+    return ToJson(std::nullopt, "", request.id);
+  return ToJson(*route_info,
+                *map_renderer_->RenderRoute(*route_info),
```
Comment (foxfrikses) [inline src/request_processor.cpp:149]: What if there is some logical error and RenderRoute returns nullopt?

File: tests/map_renderer_test.cpp [hunk 2]
```diff
@@ -602,25 +602,13 @@ TEST(TestMapRenderer, TestRenderRoute) {
                       NEXT_POINT(194.033, 207.133) NEXT_POINT(170, 225.7)
                       NEXT_POINT(194.033, 207.133) NEXT_POINT(242.1, 170)
                       ROUTE_END()
-                      ROUTE("purple") POINT(170, 225.7)
```
Comment (foxfrikses) [inline tests/map_renderer_test.cpp:605]: How is this related to this commit?

File: src/request_processor.cpp (hunk 1 above)
Comment (siarheishut) [reply src/request_processor.cpp:149]: Fixed.

File: tests/map_renderer_test.cpp (hunk 2 above)
Comment (siarheishut) [reply tests/map_renderer_test.cpp:605]: Removed.

File: src/request_processor.cpp [hunk 3]
```diff
@@ -63,14 +63,16 @@ json::Dict ToJson(std::optional<StopResponse> response, int id) {
   return result;
 }
 
-json::Dict ToJson(std::optional<RouteResponse> response, int id) {
+json::Dict ToJson(std::optional<RouteResponse> response, std::string map,
```
[the remark this answers is not in the document]
Comment (siarheishut) [reply src/request_processor.cpp]: Done.

File: tests/request_processor_test.cpp [hunk 4]
```diff
@@ -144,14 +147,15 @@ TEST(TestProcessRequests, TestRouteResponseToJson) {
...
               {"request_id", 7421097}
           }
       },
   };
 
-  for (auto &[name, response, id, want] : test_cases) {
-    json::Dict got = rm::ToJson(response, id);
-    EXPECT_EQ(want, got);
+  for (auto &[name, response, map, id, want] : test_cases) {
+    json::Dict got = rm::ToJson(response, map, id);
```
Comment (foxfrikses) [inline tests/request_processor_test.cpp:157]: How about testing nullopt map?

Comment (siarheishut) [reply tests/request_processor_test.cpp:201]: Added.

**Review by foxfrikses — APPROVED** [review]
LGTM
